frontends/p4/structInitializers.cpp

```diff
@@ -28,26 +28,129 @@ convert(const IR::Expression* expression, const IR::Type* type) {
         auto si = new IR::IndexedVector<IR::NamedExpression>();
         if (auto le = expression->to<IR::ListExpression>()) {
             size_t index = 0;
+            auto l = le->size();
             for (auto f : st->fields) {
+                if (index >= l) {
+                    break;
+                }
                 auto expr = le->components.at(index);
                 auto conv = convert(expr, f->type);
                 auto ne = new IR::NamedExpression(conv->srcInfo, f->name, conv);
                 si->push_back(ne);
                 index++;
             }
+            if (le->defaultInitializer) {
+                modified = true;
+                IR::Expression* expr;
+                for (; index < st->fields.size(); index++) {
+                    auto f = st->fields.at(index);
+                    if (f->type->is<IR::Type_Bits>()) {
+                        auto conv = new IR::Constant(0);
+                        expr = conv;
+                        auto ne = new IR::NamedExpression(expression->srcInfo, f->name, expr);
+                        si->push_back(ne);
+                    } else if (f->type->is<IR::Type_Boolean>()) {
+                        auto conv = new IR::BoolLiteral(false);
+                        expr = conv;
+                        auto ne = new IR::NamedExpression(expression->srcInfo, f->name, expr);
+                        si->push_back(ne);
+                    } else if (f->type->is<IR::Type_Error>()) {
+                        auto path = new IR::Path(IR::ID(expression->srcInfo, "error"));
+                        auto typeName =  new IR::Type_Name(expression->srcInfo, path);
+                        auto id = new IR::ID(expression->srcInfo, "NoError");
+                        auto tne = new IR::TypeNameExpression(expression->srcInfo, typeName);
+                        expr = new IR::Member(expression->srcInfo, tne, *id);
+                        auto ne = new IR::NamedExpression(expression->srcInfo, f->name, expr);
+                        si->push_back(ne);
+                    } else if (auto t = f->type->to<IR::Type_Enum>()) {
+                        auto tne =  new IR::TypeNameExpression(expression->srcInfo,
+                                                               new IR::Type_Name(t->name));
+                        expr = new IR::Member(expression->srcInfo, tne, t->members.at(0)->name);
```

Comment:
Are we checking somewhere that enums have at least 1 member?
Can you add a test for this case in testdata/p4_16_errors?

Reply:
Is it possible for enum not to have at least 1 member? According to parser it should be a syntax error, am I right?